syncapi/streams/stream_pdu.go

```diff
@@ -462,6 +475,17 @@ func (p *PDUStreamProvider) getJoinResponseForCompleteSync(
 	recentEvents := p.DB.StreamEventsToEvents(device, recentStreamEvents)
 	stateEvents = removeDuplicates(stateEvents, recentEvents)
 
+	events := recentEvents
+	// Only apply history visibility checks if the response is for joined rooms
+	if !isPeek {
+		events, err = applyHistoryVisibilityFilter(ctx, p.DB, p.rsAPI, roomID, device.UserID, recentEvents)
```

Comment:
Does this mean peek rooms will always see everything regardless of the history visibility? Seems unsafe!

Reply:
This should be something for a different PR, otherwise this PR will get even larger.